packages/react-components/src/components/PromoBanner/PromoBanner.tsx

```diff
@@ -32,23 +30,42 @@ export const PromoBanner: React.FC<PromoBannerProps> = ({
   img,
   light = false,
   linkText,
-  size = 'small',
   onButtonClick,
   onClose,
   onLinkClick,
 }) => {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const [isSmallContainer, setIsSmallContainer] = React.useState(false);
```

Comment:
why not using the enum with container size (small / large) instead of 2 separate states?

Reply:
Because we have three different states of how this banner is displayed, based on the width, I think we need to have those states.

Reply:
The number of possible states in which the banner can be displayed does not directly imply the number of state pieces you need. As long as they are mutually exclusive, they can be stored in a single state value as only one of the possible variants can be set at the time. 

Additionally, the current implementation uses two boolean values, which can store 2^2 = 4 possible values, while here, we can 3. That means there is one corrupted state theoretically possible to reach, which later can cause bugs: It has both:`isSmallContainer: true` and `isLargeContainer: true`.

Using a single state with an explicit set of possible values reduces such risk of going into an impossible state completely.
You can represent it like this:
```suggestion
  const [containerSize, setContainerSize] = React.useState<'small' | 'medium' | 'large'>('medium')
```

then later in places where you have used `isSmallContainer` replace it with `containerSize === 'small'` and similarly `isLargeContainer` with `containerSize === 'large'`.